packages/indexer-common/src/indexer-management/allocations.ts

```diff
@@ -667,19 +678,17 @@ export class AllocationManager {
       allocation,
     )
 
-    // If there is not yet an indexingRule that deems this deployment worth allocating to, make one
-    if (!(await this.matchingRuleExists(logger, subgraphDeploymentID))) {
-      logger.debug(
-        `No matching indexing rule found; updating indexing rules so indexer-agent keeps the deployment synced but doesn't reallocate to it`,
-      )
-      const offchainIndexingRule = {
-        identifier: allocation.subgraphDeployment.id.ipfsHash,
-        identifierType: SubgraphIdentifierType.DEPLOYMENT,
-        decisionBasis: IndexingDecisionBasis.OFFCHAIN,
-      } as Partial<IndexingRuleAttributes>
+    // Upsert a rule so the agent keeps the deployment synced but doesn't allocate to it
+    logger.debug(
+      `Matching indexing rule found; updating indexing rules so indexer-agent keeps the deployment synced but doesn't reallocate to it`,
```

Comment:
maybe we don't need to specify `Matching indexing rule found; `?

Reply:
Agreed, I'll update.